examples/run/cuda/full_chain_algorithm.cpp

```diff
@@ -73,28 +73,37 @@ full_chain_algorithm::~full_chain_algorithm() {
 }
 
 full_chain_algorithm::output_type full_chain_algorithm::operator()(
-    const cell_container_types::host& cells) const {
-
-    // Copy the cells to the device (asynchronously).
-    cell_container_types::buffer hostCellBuffer;
-    cell_container_types::buffer deviceCellBuffer =
-        m_host2device(get_data(cells), hostCellBuffer);
+    const alt_cell_collection_types::host& cells,
+    const cell_module_collection_types::host& modules) const {
+
+    // Execute the partitioning algorithm (on the host)
+    ccl_partition_collection_types::host partitions =
+        m_partitioning(cells, modules);
+
+    // Create device copy of input collections
+    alt_cell_collection_types::buffer cells_buffer(cells.size(),
+                                                   *m_cached_device_mr);
+    m_copy(vecmem::get_data(cells), cells_buffer);
+    cell_module_collection_types::buffer modules_buffer(modules.size(),
+                                                        *m_cached_device_mr);
+    m_copy(vecmem::get_data(modules), modules_buffer);
+    ccl_partition_collection_types::buffer partitions_buffer(
+        partitions.size(), *m_cached_device_mr);
+    m_copy(vecmem::get_data(partitions), partitions_buffer);
+
+    // Synchronize assynchronous copies.
+    m_stream.synchronize();
```

Comment:
It's a pity that the clusterization lost its asynchronicity by this PR and now you need to synchronize before calling the clusterization. :frowning:

Reply:
This was my mistake. This synchronisation point was unnecessary and has been removed